Zend/zend_API.c

```diff
@@ -2046,7 +2046,17 @@ ZEND_API zend_result add_next_index_reference(zval *arg, zend_reference *ref) /*
 }
 /* }}} */
 
-ZEND_API void zend_illegal_array_offset(const zval *offset)
+void zend_illegal_empty_or_isset_offset(const zval *offset)
+{
+	zend_type_error("Cannot access offset of type %s in isset or empty", zend_get_type_by_const(Z_TYPE_P(offset)));
+}
+
+void zend_illegal_unset_offset(const zval *offset)
+{
+	zend_type_error("Cannot access offset of type %s on unset", zend_get_type_by_const(Z_TYPE_P(offset)));
+}
+
+void zend_illegal_array_offset(const zval *offset)
 {
 	zend_type_error("Cannot access offset of type %s on array", zend_get_type_by_const(Z_TYPE_P(offset)));
 }
```

Comment:
If we are going to have them in ``zend_API.c`` then at least have them next to where all the other error helper functions are defined, near the top of the file around line 400.